src/lib/Menu/ContentTypeEditAnchorMenuBuilder.php

```diff
@@ -0,0 +1,139 @@
+<?php
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace Ibexa\AdminUi\Menu;
+
+use Ibexa\AdminUi\Config\AdminUiForms\ContentTypeFieldTypesResolverInterface;
+use Ibexa\AdminUi\Menu\Event\ConfigureMenuEvent;
+use Ibexa\Contracts\AdminUi\Menu\AbstractBuilder;
+use Ibexa\Contracts\Core\Repository\Values\ContentType\ContentTypeDraft;
+use Ibexa\Core\Repository\Values\ContentType\FieldDefinition;
+use JMS\TranslationBundle\Model\Message;
+use JMS\TranslationBundle\Translation\TranslationContainerInterface;
+use Knp\Menu\ItemInterface;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+
+final class ContentTypeEditAnchorMenuBuilder extends AbstractBuilder implements TranslationContainerInterface
+{
+    public const ITEM__GLOBAL_PROPERTIES = 'content_type_edit__anchor_menu__global_properties';
+    public const ITEM__FIELD_DEFINITIONS = 'content_type_edit__anchor_menu__field_definitions';
+
+    private const ITEM_ORDER_SPAN = 10;
+
+    private ContentTypeFieldTypesResolverInterface $contentTypeFieldTypesResolver;
+
+    public function __construct(
+        MenuItemFactory $factory,
+        EventDispatcherInterface $eventDispatcher,
+        ContentTypeFieldTypesResolverInterface $contentTypeFieldTypesResolver
+    ) {
+        parent::__construct($factory, $eventDispatcher);
+
+        $this->contentTypeFieldTypesResolver = $contentTypeFieldTypesResolver;
+    }
+
+    protected function getConfigureEventName(): string
+    {
+        return ConfigureMenuEvent::CONTENT_TYPE_EDIT_ANCHOR_MENU;
+    }
+
+    /**
+     * @param array<string, mixed> $options
+     */
+    public function createStructure(array $options): ItemInterface
+    {
+        /** @var \Knp\Menu\ItemInterface|\Knp\Menu\ItemInterface[] $menu */
+        $menu = $this->factory->createItem('root');
+
+        /** @var \Ibexa\Core\Repository\Values\ContentType\ContentTypeDraft $contentTypeDraft */
+        $contentTypeDraft = $options['content_type'];
+
+        $items = [
+            $this->createMenuItem(
+                self::ITEM__GLOBAL_PROPERTIES,
+                [
+                    'attributes' => ['data-target-id' => 'Global-properties'],
+                    'extras' => [
+                        'orderNumber' => 10,
+                    ],
+                ]
+            ),
+            $this->createMenuItem(
+                self::ITEM__FIELD_DEFINITIONS,
+                [
+                    'attributes' => ['data-target-id' => 'Field-definitions'],
+                    'extras' => [
+                        'orderNumber' => 20,
+                    ],
+                ]
+            ),
+        ];
+
+        $menu->setChildren(
```

Comment:
If I remember correctly `Menu::setChildren` carried some additional side-effects (or lack thereof). I think we encountered this when dealing with menu factory not being called?

I'd say we should add each field to the menu instead.